src/DynamoCore/Graph/Nodes/DummyNode.cs

```diff
@@ -56,6 +56,13 @@ public DummyNode(int inputCount, int outputCount, string legacyName, XmlElement
             Description = GetDescription();
             ShouldDisplayPreviewCore = false;
 
+            if (OriginalNodeContent != null)
+            {
+                var legacyFullName = OriginalNodeContent.Attributes["function"];
```

Comment:
Oh, I missed this point. what if this DummyNode doesn't represent a DSFunction node? Should we initialize LegacyFullName = legacyName at the beginning?